### ISSUE TITLE
Add option to skip already download file without checking its content

### ISSUE THREAD

**Issue by stevenleeS0ht** [ytdl-org/youtube-dl#29713]
## Checklist

- [x] I'm reporting a feature request
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've searched the bugtracker for similar feature requests including closed ones

## Description

Currently, when the target download file already exist, `youtube-dl` will firstly retrieve the data from the YouTube (all mutlimedia data), and then check the content. 

It would be nice to add option which directly check the filename, if the file which has default target download filename, `youtube-dl` will quickly skip it.

This feature is useful when downloading a large group of video (from YouTube playlist). When network or transmission error occurs, it can quickly resume without cost extra data quota.

**Comment by rautamiekka** [comment]
That's simply technically not possible. Something like this was asked before, where the dev said it's just not possible.

**Comment by dima4p** [comment]
`--no-overwrites` option does not help when `-x` and `--audio-format` options are used, it starts download 

```
$ ll 20220315-Статус-Статус*
-rw-rw-r-- 1 dima dima  3288142 Mar 16 13:18 20220315-Статус-Статус\ _\ \@Екатерина\ Шульман\ и\ Максим\ Курников\ _\ 15.03.22.m4a.part
-rw-rw-r-- 1 dima dima 41648776 Mar 15 20:10 20220315-Статус-Статус\ _\ \@Екатерина\ Шульман\ и\ Максим\ Курников\ _\ 15.03.22.mp3

```

P.S. my command line starts with `youtube-dl -i -x -w --dateafter now-1month --audio-format mp3`.

**Comment by dirkf** [comment]
The download starts if the resulting file isn't already present, or, as hinted above, you can use `--download-archive ...` to record previously downloaded items without having to keep the downloaded video file.

**Comment by dima4p** [comment]
> The download starts if the resulting file isn't already present, or, as hinted above, you can use `--download-archive ...` to record previously downloaded items without having to keep the downloaded video file.

I am doing that already, thanks.

**Comment by dirkf** [comment]
yt-dl doesn't somehow remember that you created a download archive: you have to mention the file where you recorded the previous downloads in the download archive option in your command line to avoid repeat downloads.

Some users might keep several download archives, maybe one for a particular collection.

**Comment by afterdelight** [comment]
what did u mean by this. it still sending request header even though i put --download-archive and --no-overwrites options

**Comment by dirkf** [comment]
... and the ID of the video that you previously downloaded and from which the audio was extracted is recorded in the download archive index given as the value of the `--download-archive` option?

**Comment by afterdelight** [comment]
Yes of course the ids are already in the .txt file,. the problem its still sending requests and it took 1 second per file to verify.
I thinks its better to add option like --download-metadata as a file to local disk. then add another option like --verify-file-from-metadata to skip sending request to server to speed up checking. it only took secs to verify ~4000 files with wget -nc option while youtube-dl probably took ~4000 secs.

**Comment by dirkf** [comment]
The check against the archive index is done in the filter code (`--match-filter ...`) after the `info_dict` for the item has been extracted: that process may require some network activity: API calls, page fetches, ..., depending on the extractor.

An extractor, once the `id` is fully determined, could possibly create a minimal `info_dict` containing `id` and `ie_key` fields, and then invoke `self._downloader.in_download_archive() to check if the item has been indexed. But there isn't really a way for the extractor to tell yt-dl to skip a single item (an item in a playlist can just be omitted). Some extractors only discover the `id` after several web requests.

**Comment by afterdelight** [comment]
I don;t think that command is useful in my case.
Is there a way to download only files not in the archive.txt list by comparing the file names with the ones in input .txt?

**Comment by dirkf** [comment]
The archive index records the extractor key and the `id` of the downloaded item. There's no general way to tell whether a URL matches a record in the archive index without running yt-dl's extraction phase: for instance, the URL may be a playlist whose items vary over time. For the same reason you can't match a URL against a recorded filename.

The `--download-archive` mechanism does limit **downloading** (or should do) to items not previously recorded: it can't eliminate web requests that are required to identify the item by `id`.

**Comment by afterdelight** [comment]
Ahh, ok then. That's a bummer

**Comment by pukkandan** [comment]
> > --no-overwrites option does not help when -x and --audio-format options are used, it starts download

> The download starts if the resulting file isn't already present,

It is possible to fix this when user explicitly gives `--audio-format`, but it's not an elegant solution

https://github.com/yt-dlp/yt-dlp/commit/6b591b29254db07d63f4aecc7260299331efa003 (bugfix https://github.com/yt-dlp/yt-dlp/commit/da33e35b055100cc98419873f7219a778489f403
